libs/MeshKernel/include/MeshKernel/BilinearInterpolationOnGriddedSamples.hpp

```diff
@@ -81,21 +82,164 @@ namespace meshkernel
+                                                                                                                    m_yCoordinates(yCoordinates),
+                                                                                                                    m_isCellSizeConstant(false)
+    {
+        m_values = static_cast<const values_type*>(values);
+    }
+
+    template <typename values_type>
+    void BilinearInterpolationOnGriddedSamples<values_type>::Compute()
+    {
+        m_nodeResults.resize(m_mesh.GetNumNodes());
+        std::ranges::fill(m_nodeResults, constants::missing::doubleValue);
+        for (UInt n = 0; n < m_mesh.GetNumNodes(); ++n)
+        {
+            const auto node = m_mesh.m_nodes[n];
+            m_nodeResults[n] = Interpolation(node);
+        }
+
+        m_edgeResults.resize(m_mesh.GetNumEdges());
+        std::ranges::fill(m_edgeResults, constants::missing::doubleValue);
+        for (UInt e = 0; e < m_mesh.GetNumEdges(); ++e)
+        {
+            const auto& [first, second] = m_mesh.m_edges[e];
+            m_edgeResults[e] = 0.5 * (m_nodeResults[first] + m_nodeResults[second]);
+        }
+
+        m_faceResults.resize(m_mesh.GetNumFaces(), constants::missing::doubleValue);
+        std::ranges::fill(m_faceResults, constants::missing::doubleValue);
+        for (UInt f = 0; f < m_mesh.GetNumFaces(); ++f)
+        {
+            m_faceResults[f] = Interpolation(m_mesh.m_facesMassCenters[f]);
+        }
+    }
+
+    template <typename values_type>
+    double BilinearInterpolationOnGriddedSamples<values_type>::Interpolation(const Point& point) const
+    {
+
+        double fractionalColumnIndex = GetFractionalNumberOfColumns(point);
+        double fractionalRowIndex = GetFractionalNumberOfRows(point);
+
+        double columnIndexTmp;
+        fractionalColumnIndex = std::modf(fractionalColumnIndex, &columnIndexTmp);
+
+        double rowIndexTmp;
+        fractionalRowIndex = std::modf(fractionalRowIndex, &rowIndexTmp);
+
+        if (columnIndexTmp < 0 || rowIndexTmp < 0)
+        {
+            return constants::missing::doubleValue;
+        }
+
+        UInt const columnIndex = static_cast<UInt>(columnIndexTmp);
+        UInt const rowIndex = static_cast<UInt>(rowIndexTmp);
+
+        if (columnIndex + 1 >= m_numXCoord || rowIndex + 1 >= m_numYCoord)
+        {
+            return constants::missing::doubleValue;
+        }
+
+        const auto result = fractionalColumnIndex * fractionalRowIndex * getGriddedValue(columnIndex + 1, rowIndex + 1) +
+                            (1.0 - fractionalColumnIndex) * fractionalRowIndex * getGriddedValue(columnIndex, rowIndex + 1) +
+                            (1.0 - fractionalColumnIndex) * (1.0 - fractionalRowIndex) * getGriddedValue(columnIndex, rowIndex) +
+                            fractionalColumnIndex * (1.0 - fractionalRowIndex) * getGriddedValue(columnIndex + 1, rowIndex);
+        return result;
+    }
```

Comment:
Question: If the class is intantiated with short- or intType, then will this compilation fail due to mixing floating-point and integer types in the same expression?

Reply:
A possible solution could be to return double from the `getGriddedValue` function. so a static cast in this function

Reply:
Shouldn't the result also be of ValueType? Is it meaningful to return a double when the input is int for ex?

Reply:
If ValueType is int, the result will be converted implicitly (if not a compilation error) to double and so should be explicitly converted to double (static_cast), since then other tokens in the expression are of type double. It seems that the gridded value is only used in expression like this, so it may simplify the code (fewer static casts)